daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -114,6 +121,7 @@ private[lf] object Speedy {
       var localContracts: Map[V.ContractId, (Ref.TypeConName, SValue)],
       // global contract discriminators, that are discriminators from contract created in previous transactions
       var globalDiscriminators: Set[crypto.Hash],
+      var cachedContracts: Map[V.ContractId, CachedContract],
```

Comment:
What is the lifecycle of this cache? IIUC it prevents fetching the same contract more than once while evaluating a single transaction. If I'm wrong and it survives across evaluations, would it make sense (and be feasible) to apply some eviction policy to prevent this from blowing off and causing OOMs?

Reply:
It never survives a transaction (which applies to all fields in here, speedy is pretty much only alive for one transaction with the exception of scenarios and there we reset it after each submit) and the size of the cache is at most proportional to the size of the transaction we’re building (each cache entry corresponds to a fetch or an exercise). The values in the cache are also mostly shared with the actual transaction node so we don’t keep signatories twice in memory or something like that.

Reply:
In other words, given that it doesn’t grow more than the transaction and the overhead to that seems quite small given that most values are shared, I wouldn’t add an eviction policy but I’m more than happy to be proven wrong here and add one.

Reply:
Nah, if this is only for a transaction, a simple map works. Thanks!